ihontaryk/homework_8/restaurant.py

```diff
@@ -0,0 +1,84 @@
+from collections import OrderedDict
+from collections import namedtuple
+
+
+class Restaurant:
+    menu = {'first_course': 'First Course',
+            'second_course': 'Second Course',
+            'dessert': 'Dessert',
+            'beverage': 'Beverage'}
+
+    def __init__(self):
+        self.order = OrderedDict()
+
+    def _add_item(self, key, count, price):
+        if key not in self.menu:
+            raise ValueError(f'{key} is not available in the menu')
+
+        name = self.menu[key]
+
+        if name not in self.order:
+            self.order[name] = 0
+        self.order[name] += count * price
+
+    def get_client_order(self, order: tuple):
+        for item in order:
+            self._add_item(item.kind, item.count, item.price)
+
+    def calculate_total_cost(self):
+        total_cost = 0
+        result = []
+        for item, price in self.order.items():
+            total_cost += price
+            result.append(f"{item}: {price}")
+        result.append(f"Total cost: {total_cost}")
+
+        return ", ".join(result)
+
+
+class ItalianRestaurant(Restaurant):
+    menu = {'first_course': 'Pasta',
+            'second_course': 'Potato with chicken',
+            'dessert': 'Tiramisu',
+            'beverage': 'Sparkling water'}
+
+
+class SpanishRestaurant(Restaurant):
+    menu = {'first_course': 'Seafood soup',
+            'second_course': 'Paella with seafood',
+            'dessert': 'Banana mousse',
+            'beverage': 'Juice'}
+
+
+class AustrianRestaurant(Restaurant):
+    menu = {'first_course': 'Chicken soup',
+            'second_course': 'Fried pike-perch fillet',
+            'dessert': 'Milk cream strudel ',
+            'beverage': 'Lemonade'}
+
+
+Order = namedtuple("Order", ["kind", "count", "price"])
```

Comment:
OrderLine
OrderItem

but it is not an order, since get_order expects tuple of these